CanHazFunny/CanHazFunny/Jester.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+
+namespace CanHazFunny
+{
+    public class Jester
+    {
+        private readonly IJokeOutput _jokeOutput;
+        private readonly IJokeService _jokeService;
+
+        public Jester(IJokeOutput? jokeOutput, IJokeService? jokeService)
+        {
+            _jokeOutput = jokeOutput ?? throw new ArgumentNullException(nameof(jokeOutput));
+            _jokeService = jokeService ?? throw new ArgumentNullException(nameof(jokeService));
+        }
+
+        public void TellJoke()
+        {
+            string joke = _jokeService.GetJoke();
+            while (joke.Contains("Chuck") || joke.Contains("Norris")) joke = _jokeService.GetJoke();
```

Comment:
This is a personal preference, but I like to wrap the body in brackets just for readability
```suggestion
            while (joke.Contains("Chuck") || joke.Contains("Norris")) { joke = _jokeService.GetJoke(); }
```